client/src/api/block_builder/input_selection/utxo_chains/mod.rs

```diff
@@ -93,8 +96,25 @@ pub(crate) fn select_utxo_chain_inputs(
                     if !is_required_for_output && !allow_burning || is_required_for_input {
                         let nft_address = Address::Nft(NftAddress::new(nft_id));
                         let nft_required_in_unlock_condition = outputs.iter().any(|output| {
-                            // check if alias address is in unlock condition
-                            output_contains_address(output, None, &nft_address, current_time)
+                            if let Ok((required_unlock_address, unlocked_alias_or_nft_address)) = output
+                                .required_and_unlocked_address(
+                                    current_time,
+                                    // It's a new output, so the output id doesn't matter
```

Comment:
Just checking, is that really the case? Because we do use `unlocked_alias_or_nft_address` later

Reply:
Yes, if's a new output, so the output id isn't known yet, which means that it could be a new alias or nft, where the alias or nft id is set to null, then it can't unlock anything anyways since it's just being created (we could still get the hash of null as address back, but there is basically no realistic way that we have an output that requires this address then)
If it's not null, then we will use the correct alias or nft address and the provided null output id isn't used